relayer/relayer.go

```diff
@@ -116,7 +124,60 @@ func NewRelayer(
 		)
 		return nil, nil, err
 	}
-	return &r, sub, nil
+
+	// Get the latest seen block height from the database.
+	latestSeenBlockData, err := r.db.Get(r.sourceChainID, []byte(database.LatestSeenBlockKey))
+
+	// The following cases are treated as successful:
+	// 1) The database contains the latest seen block data for the chain
+	//    - In this case, we parse the block height and process warp logs from that height to the current block
+	// 2) The database has been configured for the chain, but does not contain the latest seen block data
+	//    - In this case, we save the current block height in the database, but do not process any historical warp logs
+	if err == nil {
+		// If the database contains the latest seen block data, then  back-process all warp messages from the
+		// latest seen block to the latest block
+		// This will query the node for any logs that match the filter query from the stored block height,
+		r.logger.Info("latest seen block", zap.String("block", string(latestSeenBlockData)))
+		latestSeenBlock, success := new(big.Int).SetString(string(latestSeenBlockData), 10)
+		if !success {
+			r.logger.Error("failed to convert latest block to big.Int", zap.Error(err))
+			return nil, nil, err
+		}
+
+		err = sub.ProcessFromHeight(latestSeenBlock)
+		if err != nil {
+			logger.Warn(
+				"Encountered an error when processing historical blocks. Continuing to normal relaying operation.",
+				zap.String("chainID", r.sourceChainID.String()),
+				zap.Error(err),
+			)
+		}
+		return &r, sub, nil
+	} else if errors.Is(err, database.ErrChainNotFound) || errors.Is(err, database.ErrKeyNotFound) {
```

Comment:
nit: I usually prefer to not have an "else if" if the previous "if" block ends with a return. I find it slightly clearer to start a new "if" block instead. Just a minor stylistic preference, you don't need to change this if you don't agree.

Reply:
Agreed. Updated.